rewrite-java/src/main/java/org/openrewrite/java/ChangeMethodName.java

```
@@ -100,7 +104,9 @@ public J visit(@Nullable Tree tree, ExecutionContext ctx) {
             @Override
             public J.MethodDeclaration visitMethodDeclaration(J.MethodDeclaration method, ExecutionContext ctx) {
                 J.MethodDeclaration m = super.visitMethodDeclaration(method, ctx);
-
+                if (JavaKeywordUtils.isReserved(newMethodName) || StringUtils.isBlank(newMethodName) || !newMethodName.matches(VALID_JAVA_METHOD_PATTERN)) {
+                    return m;
+                }
```

Shouldn't the validation be done in `validate()`?

---

I'm trying to reproduce the recipe;s behavior when I add this to the validate() method.
I do think this changes the behavior of this very abstract core recipe.
It used to change methods into not compiling java code.
My approach is to silently ignore changes that are breaking java (e.g. `void this_()` should not be changed to `void this()`.
I'm a bit hesitant because if I understand correctly, changing validate would cause the recipe to start throwing errors.

---

For now I expanded the condition.

---

Moved to isAcceptable